packages/app-builder-lib/src/electron/electronMac.ts

```diff
@@ -229,31 +208,34 @@ export async function createMacApp(packager: MacPackager, appOutDir: string, asa
     appPlist.ElectronAsarIntegrity = asarIntegrity
   }
 
-  const plistDataToWrite: any = {
-    [appPlistFilename]: appPlist,
-    [helperPlistFilename]: helperPlist,
-  }
   if (helperEHPlist != null) {
-    plistDataToWrite[helperEHPlistFilename] = helperEHPlist
+    await savePlistFile(helperEHPlistFilename, helperEHPlist)
   }
+
   if (helperNPPlist != null) {
-    plistDataToWrite[helperNPPlistFilename] = helperNPPlist
+    await savePlistFile(helperNPPlistFilename, helperNPPlist)
   }
+
   if (helperRendererPlist != null) {
-    plistDataToWrite[helperRendererPlistFilename] = helperRendererPlist
+    await savePlistFile(helperRendererPlistFilename, helperRendererPlist)
   }
+
   if (helperPluginPlist != null) {
-    plistDataToWrite[helperPluginPlistFilename] = helperPluginPlist
+    await savePlistFile(helperPluginPlistFilename, helperPluginPlist)
   }
+
   if (helperGPUPlist != null) {
-    plistDataToWrite[helperGPUPlistFilename] = helperGPUPlist
+    await savePlistFile(helperGPUPlistFilename, helperGPUPlist)
   }
+
   if (helperLoginPlist != null) {
-    plistDataToWrite[helperLoginPlistFilename] = helperLoginPlist
+    await savePlistFile(helperLoginPlistFilename, helperLoginPlist)
   }
 
+  await savePlistFile(appPlistFilename, appPlist)
+  await savePlistFile(helperPlistFilename, helperPlist)
+
```

Comment:
Maybe can just keep the previous logic and then do this?
```
await Promise.all(Object.entries(plistDataToWrite).map(([filename, content] => savePlistFile(filename, content)))
```

Reply:
Currently, I don't think `Promise.all` is the best choice, especially when it comes to error messages. If there are performance issues, then `Promise.all` would indeed be better.

This approach does have a slight performance overhead, but as soon as one operation fails, it stops immediately without proceeding to write others. The error message also clearly indicates which part encountered the issue.